warp10/src/main/java/io/warp10/standalone/InMemoryChunkSet.java

```diff
@@ -399,11 +399,6 @@ public int compare(Object[] o1, Object[] o2) {
         continue;
       }
       
-      // Chunk does not intersect the main range, so if we are not fetching a preboundary, ignore it
-      if (boundaryOnly && null == boundary) {
-        continue;
-      }
```

Comment:
This never happen. In the above line
```
      if (null == chunkDecoder) {
        continue;
      }
```
and the only way for `chunkDecoder` to be non-null is 
```
        if (null != this.chunks[chunk] && !(boundaryOnly && null == boundary)) {
          chunkDecoder = this.chunks[chunk].getUnsafeDecoder(false);
        }
```
which means `!(boundaryOnly && null == boundary)` so `boundaryOnly && null == boundary` is always false.